cdap-app-fabric/src/main/java/io/cdap/cdap/support/status/SupportBundleStatus.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.support.status;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Status when generating Support bundle.
+ */
+public class SupportBundleStatus {
+  /** UUID of the bundle status object describes */
+  private final String bundleId;
+  /** status of bundle collection (IN_PROGRESS/FINISHED/FAILED) */
+  private final CollectionState status;
+  /** Failed bundle describes the failure */
+  private String statusDetails;
+  /** when bundle collection was started */
+  private final long startTimestamp;
+  /** FINISHED/FAILED bundles when bundle collection was completed */
+  private long finishTimestamp;
+  // any parameters passed to start collection
+  private final SupportBundleConfiguration parameters;
+  // Array of top-level tasks for the bundle, see task structure below
+  private List<SupportBundleTaskStatus> tasks = new ArrayList<>();
```

Comment:
You are actively changing this list from multiple threads.  If you want to do this, you need a concurrent version.